builtin/credential/aws/path_config_client.go

```diff
@@ -229,6 +236,18 @@ func (b *backend) pathConfigClientCreateUpdate(ctx context.Context, req *logical
 		configEntry.STSEndpoint = data.Get("sts_endpoint").(string)
 	}
 
+	stsRegionStr, ok := data.GetOk("sts_region")
+	if ok {
+		if configEntry.STSRegion != stsRegionStr.(string) {
+			// Region is used when building STS clients. As such, all the comments
+			// regarding the sts_endpoint changing apply here as well.
+			changedCreds = true
+			configEntry.STSRegion = stsRegionStr.(string)
+		}
+	} else if req.Operation == logical.CreateOperation {
```

Comment:
I don't think this `else` branch is necessary. It will just get the default value, which is also the zero value for strings, and so it's redundant.